### PR TITLE
Add `PrefixToInfix.java` new algorithm

### PR DESCRIPTION
**Pull request by Hardvan**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/stacks/PrefixToInfix.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+                // Push the resulting infix expression back onto the stack
+                stack.push(infix);
+            } else {
+                // Push operand onto stack
+                stack.push(Character.toString(token));
+            }
+        }
+
+        return stack.pop(); // final element on the stack is the full infix expression
```
Comment (alxkm) [inline src/main/java/com/thealgorithms/stacks/PrefixToInfix.java:66]: As far as I understand, there should only be one element in the stack. Can you imagine a situation where the final stack will not have any elements, which would mean that the expression is not well formed?

Do you think it is worth adding such a constraint before return:

```
if (stack.size() != 1) {
    throw new ArithmeticException("Malformed prefix expression");
}
```

**Review by alxkm — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/stacks/PrefixToInfix.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package com.thealgorithms.stacks;
+
+import java.util.Stack;
+
+/**
+ * Prefix to Infix implementation via Stack
```
[the remark this answers is not in the document]
Comment (Hardvan) [reply src/main/java/com/thealgorithms/stacks/PrefixToInfix.java:6]: This documentation is much clearer & concise! I have implemented this change now.

File: src/main/java/com/thealgorithms/stacks/PrefixToInfix.java (hunk 1 above)
Comment (Hardvan) [reply src/main/java/com/thealgorithms/stacks/PrefixToInfix.java:66]: Yes, such an edge case is possible.
Example: For the input `+ A B C`, the final stack contents will be `["C", "(A + B)"]`, indicating `stack.size() is > 1` and an erroneous prefix expression.

I have added the suggested stack size check in the code, as well as an additional test for this:

```java
@Test
void testMalformedPrefixExpression() {
    assertThrows(ArithmeticException.class, () -> PrefixToInfix.getPrefixToInfix("+ABC"));
}
```

**Comment by Hardvan** [comment]
I have implemented the suggested changes, and added the test for `stack size != 1` before the return statement.

**Review by alxkm — APPROVED** [review]

**Review by alxkm — APPROVED** [review]

**Pull request merged**